public/app/features/trails/otel/util.ts

```diff
@@ -98,51 +99,35 @@ export function getOtelJoinQuery(otelResourcesObject: OtelResourcesObject, scene
  */
 export function getOtelResourcesObject(scene: SceneObject, firstQueryVal?: string): OtelResourcesObject {
   const otelResources = sceneGraph.lookupVariable(VAR_OTEL_RESOURCES, scene);
-  // add deployment env to otel resource filters
-  const otelDepEnv = sceneGraph.lookupVariable(VAR_OTEL_DEPLOYMENT_ENV, scene);
-
   let otelResourcesObject = { labels: '', filters: '' };
 
-  if (otelResources instanceof AdHocFiltersVariable && otelDepEnv instanceof CustomVariable) {
+  if (otelResources instanceof AdHocFiltersVariable) {
     // get the collection of adhoc filters
     const otelFilters = otelResources.state.filters;
 
-    // get the value for deployment_environment variable
-    let otelDepEnvValue = String(otelDepEnv.getValue());
-    // check if there are multiple environments
-    const isMulti = otelDepEnvValue.includes(',');
-    // start with the default label filters for deployment_environment
-    let op = '=';
-    let val = firstQueryVal ? firstQueryVal : otelDepEnvValue;
-    // update the filters if multiple deployment environments selected
-    if (isMulti) {
-      op = '=~';
-      val = val.split(',').join('|');
-    }
-
-    // start with the deployment environment
-    let allFilters = `deployment_environment${op}"${val}"`;
-    if (config.featureToggles.prometheusSpecialCharsInLabelValues) {
-      allFilters = `deployment_environment${op}'${val}'`;
-    }
-    let allLabels = 'deployment_environment';
+    let allFilters = '';
+    let allLabels = '';
 
     // add the other OTEL resource filters
     for (let i = 0; i < otelFilters?.length; i++) {
       const labelName = otelFilters[i].key;
       const op = otelFilters[i].operator;
       const labelValue = otelFilters[i].value;
 
+      if (i > 0) {
+        allFilters += ',';
+      }
```

Comment:
Should we avoid adding `','` on the last iteration?

Reply:
Oh, this one is ok! Probably could be more clearly written but we are adding the comma before the filters, so on the last one, it should look like the following.

allfilters beofre the last iteration are == filter1="1" 

then in the last loop we go
allFilters += ','
// add the comma

allfilters += the last filter

So, there is no hanging comma at the end!